lib/galaxy/tools/verify/interactor.py

```diff
@@ -251,10 +257,16 @@ def new_history(self):
         return history_json['id']
 
     @nottest
-    def test_data_path(self, tool_id, filename):
-        response = self._get("tools/%s/test_data_path?filename=%s" % (tool_id, filename), admin=True)
+    def test_data_download(self, tool_id, filename, mode='file'):
+        response = self._get("tools/%s/test_data_download?filename=%s" % (tool_id, filename), admin=True)
         assert response.status_code == 200
-        return response.json()
+        if mode == 'file':
+            return response.content
+        elif mode == 'directory':
```

Comment:
I'm not sure what the syntax for this in tool tests is, if this isn't documented we could also drop this. That would simplify this a lot.

Reply:
This is in the xsd here https://github.com/galaxyproject/Galaxy-XSD/blob/master/galaxy.xsd#L1277 and an example is here: https://galaxyproject.org/admin/tools/writing-tests/#directory-comparison

We don't seem to have any test tool that exercises this, but I guess I could add one.